parsl/dataflow/memoization.py

```diff
@@ -94,10 +98,13 @@ def id_for_memo_dict(denormalized_dict: dict, output_ref: bool = False) -> bytes
     if type(denormalized_dict) is not dict:
         raise ValueError("id_for_memo_dict cannot work on subclasses of dict")
 
-    keys = sorted(denormalized_dict)
+    # keys = sorted(denormalized_dict)  Line that sirosen commented on
+    # Proposed solution was to normalize the keys and then sort them
+    keymap = {id_for_memo(k): k for k in denormalized_dict}
+    normed_keys = sorted(keymap.values())
```

Comment:
ok, I think this is right thing to be doing - but I don't think your test case that you added really tests that something was broken before, and is now fixed by this change.

I will try to add some comments on the test case if I can figure out a stronger test case.

Reply:
Thank you , I've also realised that and was trying to think though it. Where I fell short was I corrected the implementation first, then created the test case later. I should have done vice versa.